core/node/proof_data_handler/src/tee_request_processor.rs

```diff
@@ -0,0 +1,122 @@
+use std::sync::Arc;
+
+use axum::{extract::Path, Json};
+use zksync_config::configs::ProofDataHandlerConfig;
+use zksync_dal::{tee_proof_generation_dal::TeeType, ConnectionPool, Core, CoreDal};
+use zksync_object_store::ObjectStore;
+use zksync_prover_interface::api::{
+    GenericProofGenerationDataResponse, RegisterTeeAttestationRequest,
+    RegisterTeeAttestationResponse, SubmitProofResponse, SubmitTeeProofRequest,
+    TeeProofGenerationDataRequest,
+};
+use zksync_tee_verifier::TeeVerifierInput;
+use zksync_types::L1BatchNumber;
+
+use crate::errors::RequestProcessorError;
+
+pub type TeeProofGenerationDataResponse = GenericProofGenerationDataResponse<TeeVerifierInput>;
+
+#[derive(Clone)]
+pub(crate) struct TeeRequestProcessor {
+    blob_store: Arc<dyn ObjectStore>,
+    pool: ConnectionPool<Core>,
+    config: ProofDataHandlerConfig,
+}
+
+impl TeeRequestProcessor {
+    pub(crate) fn new(
+        blob_store: Arc<dyn ObjectStore>,
+        pool: ConnectionPool<Core>,
+        config: ProofDataHandlerConfig,
+    ) -> Self {
+        Self {
+            blob_store,
+            pool,
+            config,
+        }
+    }
+
+    pub(crate) async fn get_proof_generation_data(
+        &self,
+        request: Json<TeeProofGenerationDataRequest>,
+    ) -> Result<Json<TeeProofGenerationDataResponse>, RequestProcessorError> {
+        tracing::info!("Received request for proof generation data: {:?}", request);
+
+        let l1_batch_number_result = self
+            .pool
+            .connection()
+            .await
+            .unwrap()
```

Comment:
You should be careful with unwraps, you can simply propagate the error here (same below).

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/pull/1993/commits/1a03ce0b4a6eb7cc9ab82c9492fd4835e8d63e29) ✅

I'm not fully satisfied with the way I implemented it:
https://github.com/matter-labs/zksync-era/blob/1a03ce0b4a6eb7cc9ab82c9492fd4835e8d63e29/core/node/proof_data_handler/src/tee_request_processor.rs#L45-L49

This would probably be better:
```rust
        let mut connection = self
            .pool
            .connection()
            .await
            .map_err(|e| RequestProcessorError::Sqlx(*e.inner()))?;
```
but:
```
.map_err(|e| RequestProcessorError::Sqlx(*e.inner()))?;
                                         ^^^^^^^^^^ move occurs because value has type `zksync_dal::SqlxError`, which does not implement the `Copy` traitdoes not implement the `Copy` trait
```

If you have a better idea on how to handle it, I welcome your suggestions.